EKF/SensorRangeFinder.cpp

```diff
@@ -69,36 +65,48 @@ void Ekf::updateRangeDataValidity()
 		_rng_hgt_valid = false;
 
 		if (_range_sample_delayed.quality == 0) {
-			_time_bad_rng_signal_quality = _imu_sample_delayed.time_us;
+			_time_bad_rng_signal_quality = time_delayed_us;
 
-		} else if (_imu_sample_delayed.time_us - _time_bad_rng_signal_quality > (unsigned)_params.range_signal_hysteresis_ms) {
+		} else if (time_delayed_us - _time_bad_rng_signal_quality > _range_signal_hysteresis_ms) {
 			const bool is_in_range = ((_range_sample_delayed.rng > _rng_valid_min_val)
 						  && (_range_sample_delayed.rng < _rng_valid_max_val));
 
-			if (_is_rng_tilt_ok || is_in_range) {
+			if (isTiltOk() || is_in_range) {
 				updateRangeDataStuck();
 
-				if (!_control_status.flags.rng_stuck) {
+				if (!_is_stuck) {
 					_rng_hgt_valid = true;
 				}
 			}
 		}
 	}
 }
 
-void Ekf::updateRangeDataStuck()
+void SensorRangeFinder::updateRangeDataContinuity(uint64_t time_delayed_us)
+{
+	// update range data continuous flag (1Hz ie 2000 ms)
+	/* Timing in micro seconds */
+
+	/* Apply a 2.0 sec low pass filter to the time delta from the last range finder updates */
+	float alpha = 0.5f * _dt_update;
+	_dt_last_range_update_filt_us = _dt_last_range_update_filt_us * (1.0f - alpha) + alpha *
+					(time_delayed_us - _range_sample_delayed.time_us);
+
+	_dt_last_range_update_filt_us = fminf(_dt_last_range_update_filt_us, 4e6f);
+}
+
+void SensorRangeFinder::updateRangeDataStuck()
 {
 	// Check for "stuck" range finder measurements when range was not valid for certain period
 	// This handles a failure mode observed with some lidar sensors
-	if (((_range_sample_delayed.time_us - _time_last_rng_ready) > (uint64_t)10e6) &&
-	    _control_status.flags.in_air) {
```

Comment:
Yes, because this part of the code is only executed if all the other checks (quality and range) are passing. I would also like to have the sensor not having to perform differently if on the ground or in air, we're not fusing it the same way from the outside anyway (e.g.: terrain estimator reset to ground clearance and we create fake data in the height fusion).